internal/services/containers/container_registry_cache_rule_resource.go

```diff
@@ -0,0 +1,252 @@
+					return fmt.Errorf("creating Container Registry Cache Rule %s: %+v", id, err)
+				}
+			}
+
+			metadata.SetID(id)
+
+			return r.Read().Func(ctx, metadata)
+		},
+	}
+}
+
+func (ContainerRegistryCacheRule) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			cacheRulesClient := metadata.Client.Containers.ContainerRegistryClient_v2023_07_01.CacheRules
+			ctx, cancel := timeouts.ForRead(metadata.Client.StopContext, metadata.ResourceData)
+			defer cancel()
+
+			id, err := cacherules.ParseCacheRuleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			resourceGroupName := id.ResourceGroupName
+			registryName := id.RegistryName
+
+			registryId := registries.NewRegistryID(subscriptionId, resourceGroupName, registryName)
+
+			resp, err := cacheRulesClient.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					log.Printf("[DEBUG] Container Registry Cache Rule %s was not found.", *id)
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving Container Registry Cache Rule %s: %+v", *id, err)
+			}
+
+			metadata.ResourceData.Set("name", id.CacheRuleName)
+			metadata.ResourceData.Set("container_registry_id", registryId.ID())
+
+			if model := resp.Model; model != nil {
+				if properties := model.Properties; properties != nil {
+					metadata.ResourceData.Set("source_repo", properties.SourceRepository)
+					metadata.ResourceData.Set("target_repo", properties.TargetRepository)
+					metadata.ResourceData.Set("credential_set_id", properties.CredentialSetResourceId)
+				}
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r ContainerRegistryCacheRule) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			cacheRulesClient := metadata.Client.Containers.ContainerRegistryClient_v2023_07_01.CacheRules
+			ctx, cancel := timeouts.ForRead(metadata.Client.StopContext, metadata.ResourceData)
+
+			defer cancel()
+			log.Printf("[INFO] preparing arguments for Container Registry Cache Rule update.")
+
+			id, err := cacherules.ParseCacheRuleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			parameters := cacherules.CacheRuleUpdateParameters{}
+			credentialSetId := metadata.ResourceData.Get("credential_set_id").(string)
+
+			if credentialSetId != "" {
+				parameters = cacherules.CacheRuleUpdateParameters{
+					Properties: &cacherules.CacheRuleUpdateProperties{CredentialSetResourceId: &credentialSetId},
+				}
+			} else {
+				//This is due to an issue with the Azure CacheRule API that prevents removing credentials
+				return fmt.Errorf("Error on update: credential_set_id must not be empty: %s", id)
+			}
```

Comment:
This can be handled by setting `ForceNew` on the property in a `CustomizeDiff`. Here is a code snippet of what that looks like:
https://github.com/hashicorp/terraform-provider-azurerm/blob/8023ac4a299f56cba585f9197743d4c548682c1b/internal/services/compute/gallery_application_resource.go#L336-L344